### PR TITLE
Switching between chip erase/sector erase

### PR DESCRIPTION
**Pull request by MarianSavchuk**
Switching between chip erase/sector erase by .act files and DAP Vendor Command.

### CODE REVIEW

**Comment by 0xc0170** [comment]
what are the use cases for this two?

**Comment by MarianSavchuk** [comment]
> what are the use cases for this two?

Case#1
DAPLink is created with page erase enabled. MSC programming flow will look like this:
erase_page#0 -> program_page#0 -> verify_page#0 -> erase_page#1 -> program_page#1 -> verify_page#1 -> e.t.c.
However, if programing of entire flash is required or flash content outside the programmable image is not important, then it will be much faster to do this:
chip_erase -> program_page#0 -> verify_page#0 -> program_page#1 -> verify_page#1 -> e.t.c.
In this case, temporary switching to the chip erase mode will be useful.
For example, programming 1Mb PSoC6 with enabled page erase took 60 seconds, and with chip erasing - 30 seconds.

Case#2
DAPLink is created with chip erase enabled.
There is no way to update part of the flash through programming MSC. For example, application configuration block, keys e.t.c. 
In this case, temporary switching to the page erase mode will be useful.

File: source/daplink/drag-n-drop/flash_manager.c [hunk 1]
```diff
@@ -236,6 +239,7 @@ error_t flash_manager_uninit(void)
 
 void flash_manager_set_page_erase(bool enabled)
 {
+    config_ram_set_page_erase(enabled);
```
Comment (0xc0170) [inline source/daplink/drag-n-drop/flash_manager.c]: set_erase_method and accept two methods (enums) - sector erase or chip erase wouldn't be better?

File: docs/MSD_COMMANDS.md [hunk 2]
```diff
@@ -50,6 +50,10 @@ own right.
 
 ``msd_on.act`` Mass storage device can be enabled back by this file command on bootloader mode only if there is a target flash algo.
 
+``page_off.act`` This file temporary enables page programming and chip erasing until the next restart occurred for drag and drop.
+
+``page_on.act`` This file temporary enables page programming and sector erasing until the next restart occurred for drag and drop.
```
[the remark this answers is not in the document]
Comment (MarianSavchuk) [reply docs/MSD_COMMANDS.md]: I thought the names of these files must be in accordance with 8.3 format. Thus, the names are slightly abbreviated. Or is 8.3 format no longer a requirement?

[the remark this answers is not in the document]
Comment (0xc0170) [reply docs/MSD_COMMANDS.md]: I see, then all good

File: source/daplink/drag-n-drop/flash_manager.c (hunk 1 above)
Comment (MarianSavchuk) [reply source/daplink/drag-n-drop/flash_manager.c]: If I understand you correctly, you mean the following changes:
void config_ram_set_page_erase(bool) ==> void config_ram_set_page_erase(enum)
void flash_manager_set_page_erase(bool) ==> void flash_manager_set_erase(enum)
?

**Comment by 0xc0170** [comment]
@MarianSavchuk nodes in CI are down, we will fix that and restart jobs.

**Comment by 0xc0170** [comment]
cc @brianesquilona @flit nodes are down

**Comment by 0xc0170** [comment]
Build all OK but test has not run

/morph test

**Comment by 0xc0170** [comment]
I restarted test. It failed for one target but dont see specific failure, will need to investigate

**Pull request merged**